Added weapon_tag

Ugleh opened the pull request:
I could not test out ExtEmitSound but everything worked fine when using
EmitSound. Localization might neen to be changed, depending on your
preference. Line 6 for description and Line 90 for overlay.

Ugleh committed:
Added weapon_tag

I could not test out ExtEmitSound but everything worked fine when using
EmitSound. Localization might neen to be changed, depending on your
prefrence. Line 6 for description and Line 90 for overlay.

Ugleh committed:
Registered the weapon

Ugleh committed:
Register the weapon

Register the weapon

swampservers commented on lua/sps/items/weapons.lua:
```diff
@@ -103,6 +103,14 @@ PS_WeaponProduct({
 	class = 'weapon_gauntlet'
 })
 
+PS_WeaponProduct({
```
I'm not sure this should be a shop item. I could put it in the mystery box instead (the code for that is private, I can do it)

Ugleh replied:
> 
> 
> I'm not sure this should be a shop item. I could put it in the mystery box instead (the code for that is private, I can do it)

Sounds good, want me to revert the changes or are you? BTW Probably should test the audio levels. It was fine for me but I don't know your standard.

swampservers replied:
Revert in the same PR. I assume the audio is fine, you can revise it later if you want to.

swampservers commented on lua/weapons/weapon_tag.lua:
```diff
@@ -0,0 +1,150 @@
+
```
Make this weapon un-droppable in https://github.com/swampservers/contrib/blob/master/lua/chatcommands/cinema_simple.lua line 46

Ugleh committed:
Unregistered weapon, add undroppable to cinema_simple

swampservers commented on lua/weapons/weapon_tag.lua:
```diff
@@ -0,0 +1,150 @@
+local justReloaded = 0
+local initializeTime = 0
+local tagTimer = 35
+function SWEP:Initialize()
```
According to the wiki, SWEP:Initialize is sometimes not called clientside, so this is unreliable. Can you instead do self:SetNWFloat("initTime", CurTime()) from the serverside?

Ugleh replied:
This is the only thing I am not wrapping my head around. Because Initialize sometimes doesn't run clientside should I just set initializeTime inside the if SERVER statement? If I use the code you provided is initTime a local variable for that class?

Ugleh committed:
Remove possibility of removing item from OnDrop

swampservers commented on lua/weapons/weapon_tag.lua:
```diff
@@ -0,0 +1,150 @@
+
+
+function SWEP:DrawHUD()
+local TextWidth = surface.GetTextSize("Trebuchet24")
```
can't you GetTextSize with the message string? Not all characters are the same width

Ugleh replied:
Your right. I misunderstood the GetTextSize function. Will fix.

swampservers commented on lua/weapons/weapon_tag.lua:
```diff
@@ -0,0 +1,150 @@
+	initializeTime = CurTime()
+	if SERVER then
+		timer.Simple(5,function() if IsValid(self) and self.Owner:IsFrozen() then self.Owner:UnLock() end end)
+		timer.Simple(tagTimer,function() if IsValid(self) then self:Remove() self.Owner:Kill() end end)
```
Can you make them explode? (util.Effect, see the jihad/bigbomb code but only make one explosion effect)

Ugleh replied:
Will do, appreciate all the comments.

swampservers commented on lua/weapons/weapon_tag.lua:
```diff
@@ -0,0 +1,150 @@
+end
+
+
+function TagPlayer(self,target,attacker)
```
This func name seems like it could cause a conflict, can you make it local please?

Ugleh replied:
I assumed everything encapsulated in the class was already local to the class itself. My bad.

swampservers replied:
Nah it's global. You could do SWEP:TagPlayer(target,attacker) though, that would be better anyway

swampservers commented:
This is really good overall, thank you!

Ugleh committed:
Changes recommended by Swamp

Made TagPlayer and TestTagPlayer local, used SetNWFloat for the
initTime, fixed WordBox alignment.

Ugleh commented:
> 
> 
> This is really good overall, thank you!

I believe it is all done. All the edits you asked for.

swampservers commented on lua/weapons/weapon_tag.lua:
```diff
@@ -129,16 +138,14 @@ function SWEP:PrimaryAttack( right )
 			end
 			if (target[2] < 50) then
-				TestTagPlayer(self,target[1],self.Owner)
+				self.TestTagPlayer(self,target[1],self.Owner)
```
You can call a member function with self:TestTagPlayer (colon instead of dot) and it will implicitly pass self as the first argument

Ugleh replied:
That is cool too know. Does a period still work in this case?

swampservers replied:
It works because you passed self in manually, it's just not the "ideal" way to do it
